File: 2037-minimum-number-of-moves-to-seat-everyone/2037-minimum-number-of-moves-to-seat-everyone.cpp
class Solution {
public:
    int minMovesToSeat(vector<int>& seats, vector<int>& students) {
        vector<int> isSeat(101), isStudent(101);
        for(int i=0; i<seats.size(); i++){
            isSeat[seats[i]]++;
            isStudent[students[i]]++;
        }
        int ans=0, i=0, j=0;
        while(i<101 && j<101){
            if(isSeat[i] && isStudent[j]){
                ans+=abs(i-j);
                isSeat[i]--, isStudent[j]--;
            }
            else if(isSeat[i]) j++;
            else i++;
        }
        return ans;
    }
};

File: 2037-minimum-number-of-moves-to-seat-everyone/2037-minimum-number-of-moves-to-seat-everyone.ts
function minMovesToSeat(seats: number[], students: number[]): number {
    let freqSeats:number[] = Array(101).fill(0);
    let freqStudents:number[] = Array(101).fill(0);
    for(let i=0; i<students.length; i++){
        freqSeats[seats[i]]++;
        freqStudents[students[i]]++;
    }
    let i=0, j=0, ans = 0;
    while(i<101 && j<101){
        if(freqSeats[i] && freqStudents[j]){
            let moves = Math.min(freqSeats[i], freqStudents[j]);
            ans += moves*Math.abs(i-j);
            freqSeats[i]-=moves;
            freqStudents[j]-=moves;
        }
        else if(freqSeats[i]){
            j++;
        }
        else{
            i++;
        }
    }
    return ans;
};